### PR TITLE
Sync trie optimization

### PR DESCRIPTION
**Pull request by iulianpascalau**
- optimized the synchronization of tries by tweaking the NumConcurrentTrieSyncers parameter (changed from 50 to 2000)
- added MaxHardCapForMissingNodes parameter as to not encounter OOM exceptions when syncing multiple tries in parallel 
- defined the 2 trie sync parameters in config.toml file

### CODE REVIEW

File: cmd/node/config/config.toml [hunk 1]
```diff
@@ -776,3 +776,7 @@
 
 [Logs]
     LogFileLifeSpanInSec = 86400
+
+[TrieSync]
+    NumConcurrentTrieSyncers  = 2000
+    MaxHardCapForMissingNodes = 500
```
Comment (BeniaminDrasovean) [inline cmd/node/config/config.toml:782]: Add a new line at the end of the file.

**Review by BeniaminDrasovean — APPROVED** [review]

File: cmd/node/config/config.toml (hunk 1 above)
Comment (iulianpascalau) [reply cmd/node/config/config.toml:782]: done

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

File: cmd/node/config/config.toml [hunk 2]
```diff
@@ -776,3 +776,7 @@
 
 [Logs]
     LogFileLifeSpanInSec = 86400
+
+[TrieSync]
+    NumConcurrentTrieSyncers  = 2000
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply cmd/node/config/config.toml:781]: exactly

[the remark this answers is not in the document]
Comment (raduchis) [reply cmd/node/config/config.toml:781]: ok

File: cmd/node/config/config.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply cmd/node/config/config.toml]: fixed

**Review by stefanandy — APPROVED** [review]
System Test Passed

**Pull request merged**